waku/v2/protocol/lightpush/waku_lightpush_test.go

```diff
@@ -226,3 +227,100 @@ func TestWakuLightPushAutoSharding(t *testing.T) {
 	wg.Wait()
 
 }
+
+func TestWakuLightPushCornerCases(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	testTopic := "/waku/2/go/lightpush/test"
+	testContentTopic := "/test/10/my-lp-app/proto"
+
+	// Prepare peer manager instance to include in test
+	pm := peermanager.NewPeerManager(10, 10, utils.Logger())
+
+	node1, sub1, host1 := makeWakuRelay(t, testTopic)
+	defer node1.Stop()
+	defer sub1.Unsubscribe()
+
+	node2, sub2, host2 := makeWakuRelay(t, testTopic)
+	defer node2.Stop()
+	defer sub2.Unsubscribe()
+
+	lightPushNode2 := NewWakuLightPush(node2, pm, prometheus.DefaultRegisterer, utils.Logger())
+	lightPushNode2.SetHost(host2)
+	err := lightPushNode2.Start(ctx)
+	require.NoError(t, err)
+	defer lightPushNode2.Stop()
+
+	port, err := tests.FindFreePort(t, "", 5)
+	require.NoError(t, err)
+
+	clientHost, err := tests.MakeHost(context.Background(), port, rand.Reader)
+	require.NoError(t, err)
+	client := NewWakuLightPush(nil, nil, prometheus.DefaultRegisterer, utils.Logger())
+	client.SetHost(clientHost)
+
+	host2.Peerstore().AddAddr(host1.ID(), tests.GetHostAddress(host1), peerstore.PermanentAddrTTL)
+	err = host2.Peerstore().AddProtocols(host1.ID(), relay.WakuRelayID_v200)
+	require.NoError(t, err)
+
+	err = host2.Connect(ctx, host2.Peerstore().PeerInfo(host1.ID()))
+	require.NoError(t, err)
+
+	clientHost.Peerstore().AddAddr(host2.ID(), tests.GetHostAddress(host2), peerstore.PermanentAddrTTL)
+	err = clientHost.Peerstore().AddProtocols(host2.ID(), LightPushID_v20beta1)
+	require.NoError(t, err)
+
+	msg2 := tests.CreateWakuMessage(testContentTopic, utils.GetUnixEpoch())
+
+	// Wait for the mesh connection to happen between node1 and node2
+	time.Sleep(2 * time.Second)
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		<-sub1.Ch
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		<-sub2.Ch
+	}()
+
+	var lpOptions []Option
+	lpOptions = append(lpOptions, WithPubSubTopic(testTopic))
+	lpOptions = append(lpOptions, WithPeer(host2.ID()))
+
+	// Check that msg publish has passed for nominal case
+	_, err = client.Publish(ctx, msg2, lpOptions...)
```

Comment:
We should check if published message is received by node1 and if not fail the test.

Reply:
Addressed at https://github.com/waku-org/go-waku/pull/957/commits/7973079e09f425784f47599cd7b71ae08c29d618 Please have a look.